app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,36 @@
+class FavoritesController < ApplicationController
+  before_action :require_login
+
+  def create
+    @favorite = Favorite.new(favorite_params)
+    if @favorite.save
+      redirect_to root_path
```

Comment:
It would be nice to have a flash message telling me that I have favorited this item.